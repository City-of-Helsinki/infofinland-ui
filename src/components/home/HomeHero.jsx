import Image from 'next/image'

const HomeHero = ({ title, src }) => (
  <div className="relative mb-14 h-homehero md:h-homeheroxl bg-blue-white">
    {src && <Image src={src} layout="fill" alt="" priority objectFit="cover" />}
    <h1 className="absolute bottom-0 mb-16 ml-6 md:max-w-lg text-h1 md:text-h1xl lg:text-h1xl text-white md:ms-12 ifu-hero-home__title text-shadow-home md:min-w-md">
      {title}
    </h1>
  </div>
)

export default HomeHero
