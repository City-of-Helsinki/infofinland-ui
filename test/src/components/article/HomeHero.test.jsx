/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import HomeHero from '@/components/home/HomeHero'

describe('HomeHero', () => {
  it('should render the given image inside a  Home Hero Block', () => {
    const { asFragment } = render(<HomeHero image="/hero.png" title="title" />)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
