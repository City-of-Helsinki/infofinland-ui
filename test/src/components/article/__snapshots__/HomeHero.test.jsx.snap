// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHero should render the given image inside a  Home Hero Block 1`] = `
<DocumentFragment>
  <div
    class="relative mb-14 h-homehero md:h-homeheroxl bg-blue-lighter"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=%2Fhero.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fhero.png&w=640&q=75 640w, /_next/image?url=%2Fhero.png&w=750&q=75 750w, /_next/image?url=%2Fhero.png&w=828&q=75 828w, /_next/image?url=%2Fhero.png&w=1080&q=75 1080w, /_next/image?url=%2Fhero.png&w=1200&q=75 1200w, /_next/image?url=%2Fhero.png&w=1920&q=75 1920w, /_next/image?url=%2Fhero.png&w=2048&q=75 2048w, /_next/image?url=%2Fhero.png&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAICAIAAABPmPnhAAAVqXpUWHRSYXcgcHJvZmlsZSB0eXBlIGV4aWYAAHjarZppchw7koT/4xRzhMQOHAer2dxgjj+fA1kk9ST1dJuNSmQVc8ESi4d7ZJn1P/+9zX/xL8YSTIi5pJrSw79QQ3WND+W5/+r5bZ9wft9/7X23vx43Xzc5Dnne/f0zvzfYxvH4fcPnctt/PW7Ke8aVd6DPzO+AXjM7Psyfi+S4u8dteAeq635IteSfS+3uvo/3wrOU98fnM/TXIPrb/DwQMlaakau8c8tb/5zf5a7A68f6xo87v1mUruHz44vhTUfuSjDIL9v7vD/PTwP9avz3k/mn9cf4s/Fde6/w/7Bl+ngt/fmEjX82/jHxj4n914rcrydq+1jsdyPvPcve6+6uhYRF0xtRj/lYR/dwYcfk/tyWeGV+Ip/zeVVehYAcTDWf8XRew1brsPg2Nthpm912nfdhB0sMbrnMu3MDR+lY8dlVN7z8FPSy22Vf/fQFHw63DD4L3n2txZ5565lv2MLM03Kpswwmt//1Zf7Vyf/kZfaWv62VMXG9vQ52iiyWIc/pN1fhArtfv8Vj4M/rdf/zI7AIVTwYj5kLG2xPv0P0aL9jyx8/e66LvF8fW5PnOwAmYu7IYqzHA0+yPtpkn+xcthY7FhzUWLnzwXU8YGN0k0W64H1yJrviNDf3ZHuuddElp8NgE46IPvmMb6pvOCuESPzkUIihFn0MMcYUcywm1tiSTyHFlFJOArmWfQ455pRzLrnmVnwJJZZUcimlllZd9WBgrKnmWmqtrTnTmKgxVuP6xpHuuu+hx5567qXX3gbhM8KII408yqijTTf9BCZmmnmWWWdb1iyQYoUVV1p5lVVX28Ta9jvsuNPOu+y625fXXq/+9voPvGZfr7njKV2Xv7zGUZPzZwgrOInyGR5zweLxLA8Q0E4+e4oNwclz8tlTHUkRHYuM8o2ZVh7DhWFZF7f98t235/4tv5lY/i2/uf/Lc0au+//wnMF1v/vtD16bqnPjeOxmoWz6eLKPa5orhh+qpyt/fl++j1z3ls/1e/d0/upu5uTjYOfMuZLdRofT9Hun2FYbNgNxnpOjrMB1pFJkgWMsnzR42zmwRMw/3NS9ODS2vIdhIv4MT2o71MYV++kcxw53bla/VWZt69Ml1jeG34U/mGvsxNL2sn5Hs6dSfq7SfJ/g8irPOjsDJV2vuXcNVPY4w67efN6etf/znPk+WautMzh7hmlhjFk7iIYRKB0WQ6Q2Ysx/OGZbm6a4ltnLwg7UwFbX2Xnp+2kYJYQZog3R26I3Vaq/vJvfT2Q26oeGg1eoKoRjpZp6t40ASmkuWzEju3BySVojbRMfbnQ9D483XGvVluSSXZPIxW5jsK5FgA2b8gmDVZ+85+42EIjuTLjydPJa38n1GXZbAQussCfpMsfK/R8u/tXDGl0BThL3kE1pc7mQbR1erC9QXMkNmY1YWzlS3FrdPdamrRH51pEy2keeo8eQMynVZzeb0GCzGVBIJEUt/CI0Y7czZjeP5aKjYGS389mc3fXGVg5JU+GO5IbZwpdCcQ1xrOVnn6slsGLGMTrREvLouJrc3TXaRVbsOeIeraURKOOMgc/9MOQDn8BvrNdTAwsmbDHCw5SyrYIrG5Ajs0LoirKfARJm8/OJAQ+Zx3uQD6Dx6+RS8qQsKyku8jFY57kzxj408e6ZvbjFXgjdhQUI8VZkEFKEmcciKIjYzmx1upXt7j7FmyttjHHQoIn7/uMd4IpRKzQ1RSJ+hwapGczNvLFmNjxdW4RdAA8J+pSv2afIjJvK6zzclmEaO+7erNn5cwk+GCI2G2vY2YeS4FPEVF99M8bEr3niUTCUMOz5ZA7YXGxkN3mbJmdOjbw6cWlX0kLBoHwCg42mMEp6w4A4tOkY2y5mKCfWGyDvTE8XCnI83ptANRD3tDzyXJXVld2nMuNaJQUKygYB88TFM7Z+4CUm4xfWSt3XtBL0rFwzwvJIrVTG3JPkIUF3gXngbXaRo9+O0ShFDA/SDeqAqYP8YpIeEnG6Wibh9kgYENa4rShkCtCIWHLsZAbr1pbXemRSiM+sFZnVDKWpVHhkStfT0LZJSXOVfFw2+TITwoA9ESx2R792ZEE9EyO7V0Y781Zv4guX86BOW8krzAdVarQcKFXkKDk9Qx3UQ4ATsF6ka2VRgWuJPgFkNiTv2H6NOYhzDNOes/uYmgeUY6J6Zxt2wjg4baYVQJ2JpzugwElP/OBhSrYMR/wAJ56stJiu90dZ6VIkbM/pY8MmG8qz9ri/DQHoxCJ4yzpTwAIgFF6wqMhbyNg3yPPZ/LSszdsK15mU40L2Hn9RrJ5rDDwK9hoCEb7vfJqlRZdhD3nN6nIj/oEt1h12OLDRp43jyTEOpkkD1k86kEeQp2dbQ5wOsAF4Cv2Z3YIJVVAbO4IvFIHcpC52H4lcat2k1M88AnHYrIoBW4SAWANsNciFg6CASae2VkL+YKAfUA3Ao57oXwIMeTgAscdOlTrOLUAfubahy4O0ZPceM9cR12yNLU4Cz5OcpOkq7JxDT4wVz03wdDzLlrJOMWjQG6ToXN36YRnZuQaCnGIB7wg+tUk4AbTAEVusvQ/rqUHUMbt2gJyBPgSFV4DjtUnp4qeSYuL3iYzJlJ/ZkMBwSnL/EXOATcEGiaF4ahmC4KQgWyV9czFg70J82VsX2PsoM8MyhwfRCOY1aoV82VSHI0kX+BepvgqtObj5ZC+RYv44Q5qCK4oZENe0FkI/ByAJRxNpJ35AViDmFCjCeZtz/Bw9qLQxRu8ZZkJwkm2TSf+Q9fITLHRPaUbOHFoDPFFSepwAUnK++9aetQXfTyorU55XqI4ax7YAek4wRV6k8iRekMvEWehokdohRqTf1NgAEvFGRQFN2GFaE28T34t0AhaoPzYnwgcQIWjAKfILK1U0LTCAkRZ0DSoJGQGd4QdEOHC/QANb0W2jMzvB9951wtVeJtIBLLiaYXFgQqLwEf0kGnDlgRPOgcEkfyXue02+1zJD6SVkMqJbNHMku6f2EWCVspHdd+Ro73FiVWdOQLDSiAboroJIFdKB/xZ0j3V1fEiiwV0E6sN02AO1B1qzvejJQlCwwnZoM2yVtISiggaV5bFzDJuiIF9jKSXRGLCZZurPm5gcCg72LIo4mTYTwqZ3IB1mA8IAfwrHZu+68Tq5ss9mzDl6jqnovZskjohEvISL/g4hW/xoXnhL5md8YkocgxDpkbipqlgUjwpK9QxpqkKfRakHzeFAleBCkIWXRAD7iuxbS86KdgQ9z2HNxAkd1kSrPxCaw1SqXfx/zjVEDGTA7PWc9cDbIuGXno2A64sEIXjw9yLfAVHE0hKukIFBEbKpQHGII4rABmo/nJi6Di8v8EWCnqSuCnq3IzBwdiqFgDk8W52X5U5CGw6/FW6CJCiZgZMEBAgYtcAMghnfgnhZYoOayRTiDDfWquCPkuOlY25BFaVfzHK0CEFVlSMchNQMbqoWxhAf2CYSa7muzDt6F6TxN0GwuuospZmQbM80DCxVnD1ZAlWvHSupJIkEjqziiw3ZYmj9RA9wRR3zZ4uq2VoBlTSY1TtQQeFGWICIiAc/KmmRktKpz7/drFHLOXJOLBP8m8MsB/YpGoxHomKzUpMvw0PuQh/9PDmANlUuxkGpYkp/z5jfT6X6IETwbRqlXv05KWqOpFrLMgtMyHJ9fvAu8hzdNLLHRgsxFHFxUTEUs6ew8YsqyOahoFR6VpePhY42ljIO0p+FzKL2ZwdXMbsI+jIQ5u4+MhFXYPBXKY6rFEPqeIcKLbmgynFYCYvBa0CPtVA/nF6C7LzzBGnFq75YcST5qERpwdDmPMSWQJxQYij5EDQTckcQbIPQheGQmPXWkSbc20f2QCBBoG7t5cYw2pztuoBXkfEnP4caOhBOE1tUbv3ltM42ZRguoJZDlAn4uqnFe9yp3wIPjCAU2l9Oci6Nw0S+2gjrlxZDepsKT9gmg4V+3FxyVkUP2nAL+CM9nNpy+V+eRw+6m2uPegqsfKijst3tjPRe1uPJWxjhJGoRwPvqxygeq3qW41l5FS4Og6vEz9VboMqDKY0pwfmOULO/jMJmzygiomhO0BsXOQfVCbNjowLDQ5i2sEBI/SfRYsRpTVypANeqF+peAFuoF6ACLkoyokupFwATCh6itbicGhqu167eIev7YVfE+Ly5mD/VVbZH17P7QTmBKBB7pGc3SMCE0qgdVZgJexEOhEuJ1AFQP2FX/1rnHQSKTjG5LBLUyEhGgoOtrVzXk9QTIuObQI3lpQGpBKrzjxX2R+2xJcoPsQAPwV6CglRebG0o5jNYS9xBkPFIVEaKqWBKVVPLmIRPlECqVGMEatVjEclelcdcbjnyIAIsUgXo9oMO2SiX1wOlUnCg/Ozkb52bcs9gnqDp0t9QNFQxShUOSb6CxOLqIv23l8aO1DeeS4nZKK0q8PAPK9NlNQl8BGjmJAyQAWS+2f1WtcYltSDvI74jGIktMvYg6tPGEcXRF0n+B1jusDMEJrwQMG7SSGaqNgjcJVTYRF2d6gRJnD8vJAmxHGZOKVdLFOJ4S1nIjyjxSNR+KqbUUgjye5GQIVWtBO6A3oGGSwF5DfrDnpD3X+1prkHVq6v2yd0HB0PKaO/kGzGAdPAOKYRaZVFIzjJEllRkj4pLTkSJHVH7ka2UNjAvHtOqoWCvrMqUf+t1b4P4bkS8L7yQlNAokhFSSwT157Y0LgMpN5wJ5AIdA8gewiTPqu4KVbPnUyv5tVQrm739ANSYo/jjjGGARK/auECrQHFBB0974C2JUOEc/AXxv80ndgPowuzUl6IOuiwzQAi3QWE8Aa7TIJwJhQgeFZX1rnJE7kiYLJGH+pTb1IyqLCqLjAYKExNlum6c8icjxtG4iA7UpCfxdFGsI52WYtTTtj/hCn4EWSjfqRdDiroBw9piewlmXWcNgHOw1lanVYgCO8IN31KLXYXPqwkEd8OMS3ABXIv5r33j99GGvs98nUAHYEYSGx9gQcjgkibtkxQSLiWgIEUz9MSsglTkG5oMDeMDMbjA87oADMomOYFchPrkHKj7ipwR1S3bEnuCJCqhCQ7nde+GlGZ3tuNmUhdtmxQOaJD2wRLd17goiDuBufhahHd3RtKTmqROTpAKI0YQg3gculgdwP/yo7ID3kdPJAfaq0ItW1PomArqOStBqIeZCnkkFcrr+CNPNWX7hjQwZajQSqglGQk6qOeFhBIFKocbHK187jJftwEZuJ9Fa+OlK/JOURR9hFb0Cg1TaA+Ed9C+2BAYJpW325iHaIG/+IfZLkJuCShcVgGVof7srPXBXEpPpPe5QzSVOvI9lLkMX0PBvgjy4Wx40OUPnoWiUwcD5M1b/Awl8kcYD7XyRLtWmHYdIePN28vnGg4QL2gVkanTRz3Nr8W6N0aeK5Sn1SerjO4keftkcoOIhYEgISKFMoFtxOOSjreIPLsRMXAC8HeCTFtFuQ+yN2TEJZqHlIoBr7J9Vu3ARaPG2mlT7I/DWDK2ByWeKxhkywM1Z/9YjcAI2SPSAf6Tz/o6gTkpHW/r4E1pdCzyPWzET6e8PQOISno0Nkp4yEKJUxRUsohNoj8LNpOpUv23+QhRquGrndsEbFu6EuEnUqdnSNNRxEK4D04s0dCXuvn1wWvxYgylTCIpwgP6XLfXX08h2r+cxDCokZRxlwiyt0+DaTzQY6Ca+Hi8tBXAQq2+9EmFozxqcbjTRGbf1H9p0Yhwu1tofhQgVo0Q82ndshWcceU6Ius2mBvEA2iDP69nRDwmNlrV1WzMW78/9mDQf9SPduleHbqwoM1bVvGOgvEjKcL7IAgxAb3K78aD5iWT5hMMdYC13O4aOOrS3RDcwTUKXD0mUcuEktcVis2pZUipJFfcpYJqNJgeYV45WD3FQUoc/VMwboSr4X/iLinTIR+CJZLw6gykYxmuqVcA3wE+YP61lB+t/JQks2yVWmWmpgTJQzqIeivOWB9w1xWLImPReuz17GCDSfoWTYukPvgEo77M39mzjF6el1G3/mDp4EKnIvaSiLAJU4LF2YKQB9igdQgsQFJXrAsjT4/CKNdPKQcv4BwUBbAehlDUd/Xn2ZUeXdwowkY/oqiLnWMaIfpwp5U5rs1k8A5FA++WmPRTYT8dke59gAxxvJsJyUwpWgEEMsijwOUU8lN9tW31RREIaoAHtEeVRc9PwBXuWjAviGaDlLloupjsfXAAKjTlzjpfOwBUQU1cdMOVdS93HnhwmIixIHDcHuAtYwI35ko8/LS/Lw4FiayIO0+eBAcVnEJ1Bz3ExFfDpxu2NbXIVagcA78HsZKaHNT+VoNqEAPASfT4Cumy1W8iOldEs8KmvQL7E9anISa/mqDQXosq5cVPRhcRpnQwnyu4XWXq+wHh5/lgdajU6qvP2VM9YXwGQ1UIOcbDlo90apdeetSPSf0d5R3jYKeQeFh9qWa75AIhMAlN2Ah1osWSMrNXkBLG09Xf83q22kPUk+eF40O++UBMEttCuJVIaji8t9ZPa9C8nyOWUooOWTk81n09bfVZxZUz6/tpuV0dfoMlcotu6rtZ0Vwe9eFTrKEGJVl0AC0SS98D0OMF1FRl/Gd+xlc1A7ung4VXJI6BW+PBx32umP4gVmAfQWkMGLnp9OWt8FmO7YgAysnUE4vPo2LDNANKl9ZqB1ukn+G47qaGRbP1Q/n29ZwvJwHFDaGrB1vvrWafO9Wo416ATWOeZwf+9kbf40J75mwaud6HCGOfBscZOSBq3vu0FvtjLe+d3QIfICQljfDWI+6svpZr6kfnrubQ7Qmaz4xJxaCJzRD55UGCbABo/1SRZwBLcKQO22DP6tjVcRsM5s4f4ATnWSDLwq6MNyF9P2wTfPwirLlHMcx9uxYlHasZ7C6zVaKuq6eL1q8/zc3nnv37QEWV610a9IRS9JxE8sFv8171LdR1FeWD1A22chHKxvpynkFfd60DLJf7yf4naZvJ+jYHiXoULBkvD6At5+nFpzt4kN6JiqNHtfxPZwH/rLGfUKjkQtzy2qI4PS7s5ynBY9UiTV+9kO8NqsdQT9PDnBaDHe/t7WQKf0iHKcDcp3mgC9pRerfznvvPAA3JqJzh5ikEO9H0tlM+QycQ1GYhdd8/w0CVZHfNqqd7tZvbE7lB9caUotf/edi74kQNQl4gIkcjbNIQ0cqSxOciPXyZ+pJAXbffhtAdpbvLn+AzU+wY7iGXscc58mlPYy8l7e83t31m/dsXRCzrATmj+Ju+FzpvRpg3JVj1eZytshvLjz/0nPqDH2ox4mlqAqxnna9KPPaNK/MG1nkMv9+WA+avX9KRkNTD98M7vwPi+VzwATIDlsr6gCm7Z2n2cwX804k//jBOfP5osTMSmK2OyqzmfwE8VCoaG//ZFAAAAYNpQ0NQSUNDIHByb2ZpbGUAAHicfZE9SMNAHMVfU8UiLQ52EHHIUJ0siBVx1CoUoUKoFVp1MLn0C5o0JCkujoJrwcGPxaqDi7OuDq6CIPgB4ubmpOgiJf4vKbSI8eC4H+/uPe7eAUKzyjSrZwLQdNvMpJJiLr8q9r0ijAhCEJCQmWXMSVIavuPrHgG+3sV5lv+5P0dELVgMCIjEs8wwbeIN4ulN2+C8TxxlZVklPiceN+mCxI9cVzx+41xyWeCZUTObmSeOEoulLla6mJVNjXiKOKZqOuULOY9VzluctWqdte/JXxgu6CvLXKc5ghQWsQQJIhTUUUEVNuK06qRYyNB+0sc/7PolcinkqoCRYwE1aJBdP/gf/O7WKiYmvaRwEuh9cZyPUaBvF2g1HOf72HFaJ0DwGbjSO/5aE5j5JL3R0WJHwMA2cHHd0ZQ94HIHGHoyZFN2pSBNoVgE3s/om/LA4C3Qv+b11t7H6QOQpa7SN8DBITBWoux1n3eHunv790y7vx9DdnKUma4DmQAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB+UMDQgIIEBj7jUAAAD3SURBVAjXPcm9LgRRGIDh70zGHFkymYhiaCQStuMiNAqNqFyEzgWshk4h4SZIKJQmCkJI/BRsbLFhs9bPGZxM5uDM+fk+hfCWzwv01+KusB6J6Lh1KpX8RfjfF52vvLRE1H69/zaaiJzoBqZ5C4j7N6r1Zq8fq6eP58bhen8fJ2f1yXZg75ouFzOTtYOe6ZWunT9MDdW9FOXWQm1uiRFRubkRjIxmE7PnL2Y6ORofHquf7TE+ODC/DETki6JYXVGXV513o/Snk0LtrKGzRMScc4gYhiEAMMaqquKce+8R0XsfZlkGAEmSSCmjKIrjWGvNGLPWpmn6A9QhqLNWBBu+AAAAAElFTkSuQmCC); background-position: 0% 0%;"
      />
      <noscript />
    </span>
    <h1
      class="absolute bottom-0 mb-16 ml-6 md:max-w-lg text-h1 md:text-h1xl lg:text-h1xl text-white md:ms-12 ifu-hero-home__title text-shadow-home md:min-w-md"
    >
      title
    </h1>
  </div>
</DocumentFragment>
`;
